### ISSUE TITLE
False Positive on S4462 when using ContinueWith()

### ISSUE THREAD

**Issue by Corniel** [SonarSource/sonar-dotnet#2413]
### Description
When using ` System.Threading.Tasks.Task<TResult>.ContinueWith()` there is no way to avoid the usage of `Task.Result`. The whole purpose of `ContinueWith()` is to act on the outcome of the `Task`.

### Repro steps
``` C#
using System;
using System.Threading;
using System.Threading.Tasks;

namespace SonarAnalyzer.Experiments.CSharp
{
    public class S4462
    {
        public Task<S4462> Run(Func<Task<S4462>> continuation, CancellationToken token)
        {
            return continuation().ContinueWith((action) =>
            {
                return action.Result; // Compliant, ContinueWith() enforces to do so.
            });
        }
    }
}
```
### Expected behavior
I would expect that this usage is compliant. You could argue that you would have another rule that discourages the usage of  `ContinueWith()` completely, but that is a totally different debate.
